### PR TITLE
Migrate warp precompile

### PR DESCRIPTION
**Pull request by aaronbuchwald**
This PR migrates the warp precompile package from https://github.com/ava-labs/subnet-evm/tree/v0.5.9/x/warp into Coreth.

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: plugin/evm/vm.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ import (
 	"github.com/ava-labs/coreth/params"
 	"github.com/ava-labs/coreth/peer"
 	"github.com/ava-labs/coreth/plugin/evm/message"
+	warpPrecompile "github.com/ava-labs/coreth/precompile/contracts/warp"
```
Comment (ceyonur) [inline plugin/evm/vm.go:40]: We should force import module registry as well:
```
	// Force-load precompiles to trigger registration
	_ "github.com/ava-labs/coreth/precompile/registry"
```

**Review by ceyonur — CHANGES_REQUESTED** [review]
we should force load that registry, other than that lgtm

File: plugin/evm/vm_test.go [hunk 2]
```diff
@@ -94,7 +96,7 @@ var (
...
...dUpgradeBlockTimestamp\":0},\"nonce\":\"0x0\",\"timestamp\":\"0x0\",\"extraData\":\"0x00\",\"gasLimit\":\"0x5f5e100\",\"difficulty\":\"0x0\",\"mixHash\":\"0x0000000000000000000000000000000000000000000000000000000000000000\",\"coinbase\":\"0x0000000000000000000000000000000000000000\",\"alloc\":{\"0x99b9DEA54C48Dfea6aA9A4Ca4623633EE04ddbB5\":{\"balance\":\"0x56bc75e2d63100000\"},\"0100000000000000000000000000000000000000\":{\"code\":\"0x7300000000000000000000000000000000000000003014608060405260043610603d5760003560e01c80631e010439146042578063b6510bb314606e575b600080fd5b605c60048036036020811015605657600080fd5b503560b1565b60408051918252519081900360200190f35b818015607957600080fd5b5060af60048036036080811015608e57600080fd5b506001600160a01b03813516906020810135906040810135906060013560b6565b005b30cd90565b836001600160a01b031681836108fc8690811502906040516000604051808303818888878c8acf9550505050505015801560f4573d6000803e3d6000fd5b505050505056fea26469706673582212201eebce970fe3f5cb96bf8ac6ba5f5c133fc2908ae3dcd51082cfee8f583429d064736f6c634300060a0033\",\"balance\":\"0x0\"}},\"number\":\"0x0\",\"gasUsed\":\"0x0\",\"parentHash\":\"0x0000000000000000000000000000000000000000000000000000000000000000\"}"
```
[the remark this answers is not in the document]
Comment (aaronbuchwald) [reply plugin/evm/vm_test.go:99]: This was modified so that the required key is pre-funded in `vm_warp_test.go`

File: plugin/evm/vm.go (hunk 1 above)
Comment (aaronbuchwald) [reply plugin/evm/vm.go:40]: Updated

File: accounts/abi/topics.go [hunk 3]
```diff
@@ -37,6 +37,89 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
+// packTopic packs rule into the corresponding hash value for a log's topic
+// according to the Solidity documentation:
+// https://docs.soliditylang.org/en/v0.8.17/abi-spec.html#indexed-event-encoding.
+func packTopic(rule interface{}) (common.Hash, error) {
```
[the remark this answers is not in the document]
Comment (aaronbuchwald) [reply accounts/abi/topics.go:43]: Good catch

**Review by ceyonur — APPROVED** [review]

**Review by ceyonur — APPROVED** [review]

**Review by darioush — APPROVED** [review]

**Pull request merged**